test/functional/test_framework/util.py

```diff
@@ -354,7 +354,7 @@ def get_auth_cookie(datadir, chain):
             password = split_userpass[1]
     except OSError:
         pass
-    if user is None or password is None:
+    if (user is None or password is None) and wait_for_cookie is False:
```

Comment:
Thanks for reviewing and your help. If I understand your comment correctly, I modified the conditional here not to change the interface, but to not raise when we're waiting for the cookie auth credentials. When they do become available, it still returns the auth pair which we're waiting for in `wait_for_cookie_credentials()`.

Reply:
E.g. since `wait_for_cookie` is false by default, there is no change in behavior except when it's called by `wait_for_cookie_credentials`.

Reply:
Yeah, I don't understand why `wait_for_cookie` is needed? Just because it is turned off by default doesn't mean that it doesn't complicate the interface

Reply:
Hm, maybe it isn't. I modified that line before changing the conditional on line 340, because I was seeing empty creds. Good question. Re-checking.

Reply:
They should never be empty. 

Either both are a string of length of at least 1 or a ValueError is thrown

Reply:
It seems needed... I'll look more into why tomorrow, since it's getting late. Would you prefer a separate function to not change `get_auth_cookie`?

Reply:
I was seeing empty values, which is why I had to check for them at https://github.com/bitcoin/bitcoin/pull/18691/files#diff-86294e5ae5283eebdd9f98d79007a0e1R257

Reply:
I'll look further into what and why.

Reply:
Ok, I think I see what you mean.

Reply:
Good catch, fixed.